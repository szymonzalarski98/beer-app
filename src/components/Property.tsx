import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { colors, fontSize } from '../styles/helpers.style';
import theme from '../styles/theme.style';

interface Props {
  title: String;
  text: String | number;
}

const Property: React.SFC<Props> = ({ title = '', text = ''}) => {
  return (
    <View style={styles.container}>
      { title !== '' ? <Text style={[theme.text, styles.label]}>{title}</Text> : null}
      <Text style={theme.text}>{text}</Text>
    </View>
  );
};

export default Property;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    fontWeight: '600',
  },
});