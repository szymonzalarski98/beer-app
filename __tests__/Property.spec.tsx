import React from 'react';
import { render } from 'react-native-testing-library';
import Property from '../src/components/Property';

describe('Test Property component', () => {
  it ('Should renders correctly', () => {
    const { baseElement, getByText } = render(<Property title="Title" text="Test text" />);
    expect(getByText('Title')).not.toBeNull();
    expect(getByText('Test text')).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  })
})