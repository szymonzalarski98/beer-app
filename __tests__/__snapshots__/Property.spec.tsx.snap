// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Property component Should renders correctly 1`] = `undefined`;
